src/ucp/wireup/wireup.c

```diff
@@ -1060,14 +1061,41 @@ ucp_wireup_check_config_intersect(ucp_ep_h ep,
         return;
     }
 
-    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
-        ucs_assert(ep->uct_eps[lane] != NULL);
-    }
-
     old_key = &ucp_ep_config(ep)->key;
 
     ucp_ep_config_lanes_intersect(old_key, new_key, reuse_lane_map);
 
+    if (ucp_ep_config_key_has_cm_lane(new_key)) {
+        for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
+            ucs_assert(ep->uct_eps[lane] != NULL);
+        }
+
+        /* CM lane has to be re-used by the new EP configuration */
+        ucs_assert(reuse_lane_map[ucp_ep_get_cm_lane(ep)] != UCP_NULL_LANE);
+        /* wireup lane has to be selected for the old configuration */
+        ucs_assert(old_key->wireup_msg_lane != UCP_NULL_LANE);
+        /* wireup lane hasn't to be selected for the new configuration */
+        ucs_assert(new_key->wireup_msg_lane == UCP_NULL_LANE);
+
+        /* set the correct WIREUP MSG lane in case of CM */
+        reuse_lane = reuse_lane_map[old_key->wireup_msg_lane];
+        if (reuse_lane != UCP_NULL_LANE) {
+            new_key->wireup_msg_lane          = reuse_lane;
+        } else /* old wireup lane won't be re-used */ {
+            cm_wireup_ep                      = ucp_ep_get_cm_wireup_ep(ep);
```

Comment:
`/* previous wireup lane is not part of new configuration, so add it as auxiliary endpoint inside cm lane, to be able to continue wireup messages exchange */`